### PR TITLE
getter added for an array of tokens held by an owner

### LINKED ISSUE #1512: Add a getter for an array of tokens held by an owner
## Problem
I want to inherit `ERC721Full.sol` contract and add an extra getter for an array of tokens held by a given user.

## Expected behaviour
In previous versions it was possible by adding an extra function and accessing an `internal` `_ownedTokens` mapping:
````solidity
function tokensOfOwner(address _owner) external view returns (uint256[]) {
    return ownedTokens[_owner];
}
````

## Current behaviour
In version 2.x it is impossible to access a `private` `_ownedTokens` mapping.

### PR DESCRIPTION
**Pull request by Aniket-Engg**
Fixes #1512

### CODE REVIEW

File: test/token/ERC721/ERC721Full.test.js [hunk 1]
```diff
@@ -138,6 +138,14 @@ contract('ERC721Full', function ([
       });
     });
 
+    describe('tokensOfOwner', function () {
+      it('returns total tokens of owner', async function () {
+        const tokenIds = await this.token.tokensOfOwner(owner);
```
Comment (frangio) [inline test/token/ERC721/ERC721Full.test.js]: We should also be checking that the array length is exactly as expected.

I think we can use Chai's [`eql`](https://www.chaijs.com/api/bdd/#method_eql) assertion to abbreviate that and the next two lines.

```javascript
tokenIds.should.eql([firstTokenId, secondTokenId]);
```

**Comment by frangio** [comment]
@nventuro Can you confirm if you're on board with this feature?

File: test/token/ERC721/ERC721Full.test.js (hunk 1 above)
Comment (Aniket-Engg) [reply test/token/ERC721/ERC721Full.test.js]: Ids are returned as bignumber. Is there any related assertion for that?

**Comment by nventuro** [comment]
@frangio Yup, I think we should move forward with this.

File: test/token/ERC721/ERC721Full.test.js (hunk 1 above)
Comment (frangio) [reply test/token/ERC721/ERC721Full.test.js]: Hm, I see the problem. Then I'd say just add the check for length. I think the following should work.

```suggestion
        const tokenIds = await this.token.tokensOfOwner(owner);
        tokenIds.length.should.equal(2);
```

**Comment by frangio** [comment]
I don't know why the coveralls check isn't being reported, but it stayed at 100%.

Thanks @Aniket-Engg!

**Comment by imsys** [comment]
This was removed on https://github.com/OpenZeppelin/openzeppelin-contracts/commit/bd0778461da82364aebd2d763bb338795c5f9fa7 😢

**Comment by Apdlrcjafg19** [comment]
> Fixes #1512

**Pull request merged**